### PR TITLE
fix!: Improve golang documentation for types and copyright header

### LINKED ISSUE #45: Add space between copyright and package name
## Expected Behavior

A space between copyright and package name.

## Actual Behavior

There is no space today

## Details

There should be a space between the copyright and the package name or else the copyright shows up in godoc: see https://pkg.go.dev/github.com/googleapis/google-cloudevents-go/cloud/audit/v1 as an example.

### LINKED ISSUE #46: Have struct docs conform with godoc conventions
## Expected Behavior

```go
// Foo: something something something about foo.
type Foo struct {}
```

## Actual Behavior

```go
// Something something something about foo.
type Foo struct {}
```

## Details

It would be nice if the generated docs could follow godoc conventions. For example Auth could start with: “Auth: The third ...”

### LINKED ISSUE #48: Remove marshaller/unmarshaller functions
## Expected Behavior

Users of this library should work with the [json](https://pkg.go.dev/encoding/json) package directly.

## Actual Behavior

There are non-standard custom marshaller functions/methods. [ex1](https://pkg.go.dev/github.com/googleapis/google-cloudevents-go/cloud/audit/v1#UnmarshalLogEntryData) [ex2](https://pkg.go.dev/github.com/googleapis/google-cloudevents-go/cloud/audit/v1#LogEntryData.MarshalLogEntryData)

## Details

I don’t think there is a need from the customer marshallers/unmarshallers. If anything special is needed that is not directly supported by the json package I would instead [implement a custom JSON marshaller](https://pkg.go.dev/encoding/json#example-package-CustomMarshalJSON) that is respected by the json package.

### PR DESCRIPTION
**Pull request by grant**
## fix!: Improve golang documentation for types and copyright header

Based on a mini Golang friction log, this PR addresses many GitHub issues filed in this repo regarding Go usability and style.

This is a breaking change (can use semantic commit message when merging).

### Details

- Adds newline between copyright header and package declaration
  - Fixes: https://github.com/googleapis/google-cloudevents-go/issues/45
- Prefixes each struct with the name per golang convention
  - Fixes: https://github.com/googleapis/google-cloudevents-go/issues/46
- Removes custom unmarshaler functions
  - Fixes: https://github.com/googleapis/google-cloudevents-go/issues/48
- Updates main README documentation and reference documentation taking into account the above updates

### CODE REVIEW

File: cloud/firestore/v1/document_event_data.go [hunk 1]
```diff
@@ -11,18 +11,17 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
-package firestore
 
-import "encoding/json"
+package firestore
 
-// The data within all Firestore document events.
+// DocumentEventData: The data within all Firestore document events.
```
Comment (grayside) [inline cloud/firestore/v1/document_event_data.go:17]: nit: This is more commonly used in a sentence form, such as:

```suggestion
// DocumentEventData type represents the data in a Firestore document event.
```

File: README.md [hunk 2]
```diff
@@ -41,7 +42,8 @@ func main() {
     "subscription": "projects/myproject/subscriptions/mysubscription"
   }`)
 
-	e, err := pubsub.UnmarshalMessagePublishedData(data)
```
Comment (grayside) [inline README.md:44]: My preference for the custom function was in the context of performing more actions beyond just the base unmarshalling, e.g., handling the base64 decoding. Agreed that a single "task" wrapped in this function is not helpful, but if we might add this in the future it's worth considering what usage pattern we want to encourage.

**Review by grayside — APPROVED** [review]
No blocking considerations, but I commented with a style nit and point out a reason to possibly keep the unmarshalling method.

File: README.md (hunk 2 above)
Comment (grant) [reply README.md:44]: OK.
Need a full feature request for this custom unmarshaling idea. To me, it seems like there will be minimal benefit for this library to add decorator functionality than providing a tighter interface.

base64 decoding will come with https://github.com/googleapis/google-cloudevents-go/issues/49

File: cloud/firestore/v1/document_event_data.go (hunk 1 above)
Comment (grant) [reply cloud/firestore/v1/document_event_data.go:17]: OK.

I don't know how to modify the proto descriptions to conform to the style of every programming language comment form. This change to the post-proto descriptions seemed like a reasonable compromise.

**Pull request merged**